### PR TITLE
fix avgdt variable and add nthreads value to gcomp

### PR DESCRIPTION
**Pull request by jedwards4b**
### Description of changes
Fixes the avgdt variable to output in correct units of wallclock seconds per simulation day.

In the CESM driver only, add the nthreads value from nuopc.runconfig to each components attributes. Since the CESM driver is include with CMEPS - this is in fact a CMEPS change but only relevant to the driver.

### Specific notes

Contributors other than yourself, if any:

CMEPS Issues Fixed (include github issue #):

Are changes expected to change answers?
 - [X] bit for bit
 - [ ] different at roundoff level
 - [ ] more substantial 

Any User Interface Changes (namelist or namelist defaults changes)?
 - [ ] Yes
 - [X] No

Testing performed if application target is CESM:(either UFS-S2S or CESM testing is required):
- [X] (required) CIME_DRIVER=nuopc scripts_regression_tests.py
   - machines:
   - details (e.g. failed tests):
- [X] (required) CESM testlist_drv.xml
   - machines and compilers: cheyenne intel, baseline is apr05
   - details (e.g. failed tests):
- [ ] (optional) CESM prealpha test
   - machines and compilers
   - details (e.g. failed tests):

Testing performed if application target is UFS-S2S:
- [ ] (required) UFS-S2S testing
   - description:
   - details (e.g. failed tests):

Hashes used for testing:
- [ ] CESM:
  - repository to check out: https://github.com/ESCOMP/CESM.git
  - branch: nuopc_dev
  - hash:
- [ ] UFS-S2S, then umbrella repository to check out and associated hash:
  - repository to check out:
  - branch:
  - hash:

### CODE REVIEW

**Comment by mvertens** [comment]
@jedwards4b - thank you for fixing this. I will accept this PR once you document the testing that was done.

**Comment by mvertens** [comment]
@jedwards4b - also since this PR effect code that is used also by NEMS, we need to include @DeniseWorthen as a reviewer.

**Comment by jedwards4b** [comment]
@mvertens because the testing requirements outlined above are onerous I will wait and add the other changes required for threading before running tests and submitting for review.

**Comment by DeniseWorthen** [comment]
It's not clear to me if you want a review/testing now, or after you do more work on this PR. I'm assuming that you haven't finalized the changes yet but please let me know if otherwise.

**Comment by jedwards4b** [comment]
@DeniseWorthen thanks for checking - there is new code in esmf development that this PR depends on, we'll hold off until that goes into a beta tag (required for a change in the cesm driver and will not affect you).

**Review by mvertens — APPROVED** [review]

**Comment by mvertens** [comment]
@jedwards4b - please summarize your testing and related hashes. I have assigned myself to do the merge.

File: drivers/cime/esm.F90 [hunk 1]
```diff
@@ -887,7 +897,7 @@ subroutine esm_init_pelayout(driver, maxthreads, rc)
     logical, allocatable           :: comp_iamin(:)
     character(len=5)               :: inst_suffix
     character(CL)                  :: cvalue
-    logical                        :: found_comp 
+    logical                        :: found_comp
```
[the remark this answers is not in the document]
Comment (DeniseWorthen) [reply drivers/cime/esm.F90:903]: I've been planning on removing this as a driver attribute when cleaning up the nems driver.

**Pull request merged**